src/duration/__tests__/format.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019 JS Foundation and other contributors
+// Copyright 2024 YANDEX LLC
+
+import {duration} from '..';
+import {settings} from '../../settings';
+
+const dur = () =>
+    duration({
+        years: 1,
+        months: 2,
+        weeks: 1,
+        days: 3,
+        hours: 4,
+        minutes: 5,
+        seconds: 6,
+        milliseconds: 7,
+    });
+
+//------
+// #toISOString()
+//------
+
+test('Duration#toISOString fills out every field', () => {
+    expect(dur().toISOString()).toBe('P1Y2M1W3DT4H5M6.007S');
```

Comment:
Is fractional units valid?

Reply:
Yes: `If necessary for a particular application, the lowest order components may have a decimal fraction.`